rest/client.go

```diff
@@ -0,0 +1,82 @@
+package rest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// Client is a wrapper around SAMS primitive REST API. Most likely, you want to use
+// the more robust service-to-service protobuf API.
+//
+// This API is needed when using SAMS to identify users, but not perform authorization
+// checks. e.g. the caller will handle its own authorization checks based on the identity
+// of the SAMS user. (The returned User.Subject, the SAMS account external ID.)
+type Client interface {
+	// GetUserDetails returns the basic user profile of the user whom the supplied SAMS access
+	// token belongs to.
+	//
+	// If the supplied token is invalid, malformed, or expired, the error will contain
+	// "unexpected status 401".
+	GetUserDetails(ctx context.Context, token string) (*User, error)
+}
```

Comment:
Could you unwrap this interface (and move docstrings to the implementation directly)? In our current practice, consumers should define their interface on-demand for mocking/interoperability.

Reply:
Done.